pkg/app/piped/controller/scheduler.go

```diff
@@ -528,6 +529,24 @@ func (s *scheduler) executeStage(sig executor.StopSignal, ps model.PipelineStage
 		Notifier:              s.notifier,
 	}
 
+	// Skip the stage if needed based on the skip config.
+	skip, err := s.shouldSkipStage(sig.Context(), input)
+	if err != nil {
+		lp.Errorf("failed to check whether skipping the stage: %w", err.Error())
+		if err := s.reportStageStatus(ctx, ps.Id, model.StageStatus_STAGE_FAILURE, ps.Requires); err != nil {
+			s.logger.Error("failed to report stage status", zap.Error(err))
+		}
+		return model.StageStatus_STAGE_FAILURE
+	}
+	if skip {
+		if err := s.reportStageStatus(ctx, ps.Id, model.StageStatus_STAGE_SKIPPED, ps.Requires); err != nil {
+			s.logger.Error("failed to report stage status", zap.Error(err))
+			return model.StageStatus_STAGE_FAILURE
+		}
+		lp.Infof("The stage was successfully skipped due to the skip configuration of the stage.")
```

Comment:
```suggestion
		lp.Info("The stage was successfully skipped due to the skip configuration of the stage.")
```
nits. There is no need to use formater here 👀

Reply:
Thanks, I fixed it now. [e2e35f4](https://github.com/pipe-cd/pipecd/pull/4922/commits/e2e35f46f4f0fe56cee9c4126b4e34e8f4cac64d)